src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -255,6 +261,25 @@ private boolean isContainVerifiableType(DetailAST modifiers) {
         return result;
     }
 
+    /**
+     * Checks the super type and implemented interfaces of a given type.
+     * @param typeDef class or interface for check.
+     */
+    private void visitTypeDef(DetailAST typeDef) {
+        if (isVerifiable(typeDef)) {
+            checkIdent(typeDef.findFirstToken(TokenTypes.IDENT));
+            checkTypeParameters(typeDef);
```

Comment:
Issue says we are just checking `extends` and `implements`. Don't these 2 lines do more than that?
I see the content of your issue went with Class name too, but the issue title doesn't agree too.

@romani Should `IllegalType` really extend to defined class names too?

Reply:
Declaring a class with illegal name is meaningless. It will be impossible to use it.
But you are right, the issue should be renamed.

Reply:
@rnveach The issue renamed

Reply:
The issue renamed back to original, type name validation was removed.